src/freenas/etc/rc.conf.local

```diff
@@ -30,6 +29,33 @@
 . /etc/rc.freenas
 
 
+# Make our calls to middleware once, and store the results in variables.
+# Reference these variables in script to prevent from having to make \
+# duplicate middleware calls.
+
+is_licensed()
+{
+	local is_licensed=1
```

Comment:
I must be mussing something here but how is this caching is_licensed and ha_status? is_licensed and ha_status vars are local and not storing anything.

Reply:
You're right. Weekend work made me not pay attention. My apologies. I'll commit the (hopefully) final changes.